### PR TITLE
Barsign Wallmounts and Cargo Pack

### PR DESCRIPTION
**Pull request by Rhials**
## About The Pull Request

My apologies to Leif.Nobody, whose contest I was unable to submit a barsign sprite for. Hopefully this contribution will suffice.

This gives bar signs their own wallmount, allowing for them to be deconstructed, moved, or replaced from a cargo pack.

**BEHOLD: MULTI-BARSIGN ACTION!**

image

The cargo pack is expensive, because bar signs are SPECIAL and shouldn't be thrown around the station willy-nilly.

Now that bar signs are replaceable, they have been made fully destructible and will be destroyed after taking enough damage, rather than just being disabled.

They can also be removed via opening the panel and wrenching (for 10 seconds, so the bartender has time to gun down sign thiefs!)
## Why It's Good For The Game

With all the recent barsign related PRs, they could definitely use with a tad more depth on the code side. 

Honestly, I just saw the "Bargonia" bar sign and realized "there's no way a cargo bar could actually set this up as a decoration" and set out to fix that.
## Changelog
:cl: Rhials
qol: Bar signs can now be purchased from cargo. Neat!
qol: Bar signs can now be deconstructed with a screwdriver and wrench.
qol: Bar signs can now be smashed to pieces, rather than just disabled.
/:cl:

### CODE REVIEW

**Comment by BlueMemesauce** [comment]
Barsigns rename the area they're in when changed. I would recommend changing this functionality so cargo can't rename all the rooms in the station "Maid Cafe"

File: code/game/machinery/barsigns.dm [hunk 1]
```diff
@@ -129,6 +134,18 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/barsign, 32)
...
+	. = ..()
+	if(!panel_open)
+		balloon_alert(user, "open the panel first!")
+		return FALSE
+
+	tool.play_tool_sound(src)
+	if(do_after(user, (10 SECONDS), target = src))
+		tool.play_tool_sound(src)
+		deconstruct(disassembled = TRUE)
+		return TOOL_ACT_TOOLTYPE_SUCCESS
```
Comment (san7890) [inline code/game/machinery/barsigns.dm:147]:
```suggestion
	if(!do_after(user, (10 SECONDS), target = src))
		return FALSE

	tool.play_tool_sound(src)
	deconstruct(disassembled = TRUE)
	return TOOL_ACT_TOOLTYPE_SUCCESS
```

**Review by san7890 — CHANGES_REQUESTED** [review]

**Review by Wallemations — APPROVED** [review]
Icon looks fine.

Additional question:

How viable would it be for finding a way to link multiple barsigns to each other, so changing one changes the rest?
It would help with the mild annoyance of finding all the different signs in each map and manually swapping each of them.

Don't worry about doing this yourself, it was more a question for anyone who knows anything about code more than I do.

**Comment by Rhials** [comment]
> Barsigns rename the area they're in when changed. I would recommend changing this functionality so cargo can't rename all the rooms in the station "Maid Cafe"

Addressed, barsigns now only rename the area they occupy if they were built into the map.

**Comment by san7890** [comment]
hm i liked the potential interaction of having bar signs easily change names of areas because that is funny to me. maybe if the panel is open and it's "registered" on the station's blueprints (hardmatch to the ones the CE has) it'll retain its area-name changing properties? figure i should mention it now while this mechanic is still fluid

**Comment by Wallemations** [comment]
> hm i liked the potential interaction of having bar signs easily change names of areas because that is funny to me. maybe if the panel is open and it's "registered" on the station's blueprints (hardmatch to the ones the CE has) it'll retain its area-name changing properties? figure i should mention it now while this mechanic is still fluid

That kinda negates the blueprints tbh

**Comment by san7890** [comment]
it doesn't negate the blueprints if you're limited to only a certain number of area names

**Pull request merged**